src/averageAges.js

```diff
@@ -37,7 +40,15 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const onlyWomen = people.filter(person => (withChildren === undefined)
+    ? person.sex === 'f'
+    : person.sex === 'f' && people.find(someone => someone.mother
+       === person.name));
+
+  const womenAverageAge = onlyWomen.reduce((a, b) => (a + (b.died - b.born)), 0)
```

Comment:
let us move logic of looking of looking for average to separate function, to prevent code duplicating

Reply:
Hi, Alexandra! Thanks for review and recommendation! 
Done.